src/internal/base-component/__tests__/metrics.test.ts

```diff
@@ -262,120 +252,36 @@ describe('Client Metrics support', () => {
+          const metrics = new Metrics('pkg', testCase[0]);
+          metrics.logComponentUsed('DummyComponentName', { props: {} });
+          checkMetric(`awsui_DummyComponentName_u${testCase[1][0]}`, {
             o: 'main',
-            s: 'pkg',
-            t: 'default',
-            a: 'used',
+            t: 'unknown',
             f: 'react',
             v: testCase[1][1],
-            c: undefined,
+            a: 'used',
+            s: 'DummyComponentName',
+            c: { props: {} },
           });
         });
       });
     });
   });
 
-  describe('sendMetricObjectOnce', () => {
-    test('logs a metric only once if it is the same object', () => {
-      const metricObj = {
-        source: 'pkg',
-        action: 'used' as const,
-        version: '5.0',
-      };
-
-      metrics.sendMetricObjectOnce(metricObj, 1);
-      metrics.sendMetricObjectOnce(metricObj, 1);
-      expect(window.panorama).toHaveBeenCalledTimes(1);
-    });
-    test('logs metric for each different version if same source and action', () => {
-      metrics.sendMetricObjectOnce(
-        {
-          source: 'pkg1',
-          action: 'used',
-          version: '5.0',
-        },
-        1
-      );
-      metrics.sendMetricObjectOnce(
-        {
-          source: 'pkg1',
-          action: 'used',
-          version: '6.0',
-        },
-        1
-      );
-      expect(window.panorama).toHaveBeenCalledTimes(2);
-    });
-    test('logs a metric multiple times if same source but different actions', () => {
-      metrics.sendMetricObjectOnce(
-        {
-          source: 'pkg2',
-          action: 'used',
-          version: '5.0',
-        },
-        1
-      );
-      metrics.sendMetricObjectOnce(
-        {
-          source: 'pkg2',
-          action: 'loaded',
-          version: '5.0',
-        },
-        1
-      );
-      expect(window.panorama).toHaveBeenCalledTimes(2);
-    });
-  });
-
-  describe('initMetrics', () => {
-    test('sets theme', () => {
-      const metrics = new Metrics('dummy-package', 'dummy-version');
-      metrics.initMetrics('dummy-theme');
-
-      // check that the theme is correctly set
-      metrics.sendMetricObject(
-        {
-          source: 'pkg',
-          action: 'used',
-          version: '5.0',
-        },
-        1
-      );
-      checkMetric(`awsui_pkg_d50`, {
-        o: 'main',
-        s: 'pkg',
-        t: 'dummy-theme',
-        a: 'used',
-        f: 'react',
-        v: '5.0',
-        c: undefined,
-      });
-    });
-  });
-
   describe('logComponentUsed', () => {
     test('logs the usage of the given component', () => {
       metrics.logComponentUsed('DummyComponentName', { props: {} });
       checkMetric(`awsui_DummyComponentName_d10`, {
         o: 'main',
-        s: 'DummyComponentName',
         t: 'default',
-        a: 'used',
         f: 'react',
         v: '1.0',
+        a: 'used',
+        s: 'DummyComponentName',
         c: { props: {} },
       });
```

Comment:
The order of keys matters, because there is JSON.stringify down the line and we compare the serialized strings